src/StoragelessSession/Session/DataScope.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace StoragelessSession\Session;
+
+class DataScope
+{
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @var \DateTimeImmutable
+     */
+    private $expirationTime;
+
+    /**
+     * @var bool
+     */
+    private $isModified = false;
+
+    /**
+     * @var mixed[]
+     */
+    private $data = [];
+
+    /**
+     * @param string                  $value
+     * @param \DateTimeImmutable|null $dateTimeImmutable
+     */
+    public function __construct(string $value, \DateTimeImmutable $dateTimeImmutable = null)
+    {
+        $this->value          = $value;
```

Comment:
If the current date is greater than the given `$dateTimeImmutable`, then we should clear all given data